### PR TITLE
Only initialize losses if actually using them

### PR DESCRIPTION
**Pull request by joeyballentine**
I noticed when I started training ESRGAN using this, that the checkpoints were gigantic (almost 3 gigabytes) and it also downloaded a bunch of models and extra stuff for losses I wasn't using. Upon further investigation, both of these were because it was still initializing the losses in the AllLoss class regardless of if I had them configured to use.

This PR makes it only initialize the loss classes if the weight is > 0. 

This results in checkpoints of only 732mb (2GB less) as well as less vram usage (though vram usage slowly climbs back up during training).

I also noticed while doing this that SoftMarginLoss was missing from the config, so I added that as well.

### CODE REVIEW

**Comment by styler00dollar** [comment]
> Upon further investigation, both of these were because it was still initializing the losses in the AllLoss class regardless of if I had them configured to use.

The main problem is, that if one would actually load a `.ckpt` file and then add/remove a loss that had weights, it would result in a weight mismatch and thus crashing the code. It's not like I never thought about this, but I don't know if there is a good solution for this. Either everything gets saved and all loss can be toggled after resuming training or one would need to start from iteration 0 and load pretrain `.pth` every time a change happens. Not sure if lightning allows to partially load. It's also the same reason why the discriminator can't be changed during training, since the `.ckpt` will load the old weights into the discriminator and a mismatch happens.

> SoftMarginLoss was missing

Huh, ok that was overlooked.

**Comment by styler00dollar** [comment]
Merged changes [here](https://github.com/styler00dollar/Colab-traiNNer/commit/025badbdcc924be6bc0862ac030b50b30fe378c2) because `model.load_from_checkpoint(cfg["path"]["checkpoint_path"], strict=False)` allows partial state dict loading. Thanks.
